controllers/object_controls.go

```diff
@@ -1388,7 +1388,11 @@ func TransformDevicePlugin(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpe
 	// update env required for CDI support
 	if config.CDI.IsEnabled() {
 		setContainerEnv(&(obj.Spec.Template.Spec.Containers[0]), CDIEnabledEnvName, "true")
-		setContainerEnv(&(obj.Spec.Template.Spec.Containers[0]), DeviceListStrategyEnvName, "envvar,cdi-annotations")
+		if config.Operator.DevicePluginCDIDevicesFeatureEnabled() {
+			setContainerEnv(&(obj.Spec.Template.Spec.Containers[0]), DeviceListStrategyEnvName, "cdi-cri")
+		} else {
+			setContainerEnv(&(obj.Spec.Template.Spec.Containers[0]), DeviceListStrategyEnvName, "envvar,cdi-annotations")
+		}
 		setContainerEnv(&(obj.Spec.Template.Spec.Containers[0]), CDIAnnotationPrefixEnvName, "nvidia.cdi.k8s.io/")
```

Comment:
This block is also not relevant when using `cdi-cri`.

Reply:
Do you mean to say that `useDevicePluginCDIDevicesFeature` should be stronger than `cdi.enabled`? The current design and implementation is that `useDevicePluginCDIDevicesFeature` does nothing unless `cdi.enabled` is set.